tests/test_oauth2client.py

```diff
@@ -200,12 +201,20 @@ def test_create_scoped(self):
                      credentials.create_scoped(['dummy_scope']))
 
   def test_get_environment_gae_production(self):
+    sys.modules['google'] = object()
+    sys.modules['google.appengine'] = object()
     os.environ['SERVER_SOFTWARE'] = 'Google App Engine/XYZ'
     self.assertEqual('GAE_PRODUCTION', _get_environment())
```

Comment:
Possibly, and it could lead to making it difficult to determine which test is causing this. I can re-work this as a context manager if desired.